source/portable/NetworkInterface/libslirp/MBuffNetifBackendLibslirp.c

```diff
@@ -588,8 +593,8 @@ static int lSlirpGetREventsCallback( int lIdx,
     configASSERT( pxCtx );
 
     configASSERT( lIdx >= 0 );
-    configASSERT( lIdx < pxCtx->nfds );
-    configASSERT( lIdx < pxCtx->xPollFdArraySize );
+    configASSERT( lIdx < ( int ) pxCtx->nfds );
```

Comment:
Are these casts safe? When nfds or xPllFdArraySize  is larger than an int this cast will cause it to be truncated and a value larger than lIdx may evaluate to being smaller here?

The C standard only requires int to be at least 16 bits but size_t (in the second case) has to be large enough to hold the size of the largest allocatable object so it could be larger than int ...

Reply:
I don't see a problem on first glance with this. It's probably better to move the asserts after the conversion to nfds_t though.